frontend/src/container/AdvancedSearch.jsx

```diff
@@ -0,0 +1,303 @@
+      this.setHeadTitle()
+    }
+  }
+
+  setHeadTitle = () => {
+    const { props } = this
+    const headTitle = buildHeadTitle(
+      [`${props.t('Search results')} : ${parseSearchUrl(qs.parse(props.location.search)).searchedKeywords}`]
+    )
+
+    props.dispatch(setHeadTitle(headTitle))
+  }
+
+  loadSearchUrl = async () => {
+    const { props } = this
+    const searchObject = parseSearchUrl(qs.parse(props.location.search))
+    const FIRST_PAGE = 1
+
+    const fetchGetSearchedKeywords = await props.dispatch(getSearchedKeywords(
+      searchObject.contentTypes,
+      searchObject.searchedKeywords,
+      FIRST_PAGE,
+      (searchObject.numberResultsByPage * searchObject.currentPage),
+      searchObject.showArchived,
+      searchObject.showDeleted,
+      searchObject.showActive
+    ))
+
+    switch (fetchGetSearchedKeywords.status) {
+      case 200:
+        props.dispatch(setSearchedKeywords(searchObject.searchedKeywords))
+        props.dispatch(setSearchResultsList(fetchGetSearchedKeywords.json.contents))
+        props.dispatch(setCurrentNumberPage(searchObject.currentPage))
+        props.dispatch(setNumberResultsByPage(searchObject.numberResultsByPage))
+        this.setState({ totalHits: fetchGetSearchedKeywords.json.total_hits })
+        break
+      default:
+        props.dispatch(newFlashMessage(props.t('An error has happened'), 'warning'))
+        break
+    }
+  }
+
+  getContentName = (content) => {
+    // FIXME - GB - 2019-06-04 - we need to have a better way to check if it is a file than using contentType[1]
+    // https://github.com/tracim/tracim/issues/1840
+    const { props } = this
+    let contentName = ''
+
+    if (props.contentType.length > 1) {
+      contentName = content.contentType === props.contentType[1].slug ? content.filename : content.label
+    } else {
+      props.dispatch(newFlashMessage(props.t('An error has happened'), 'warning'))
+    }
+    return contentName
+  }
+
+  handleClickSeeMore = async () => {
+    const { props } = this
+    const NEXT_PAGE = props.searchResult.currentNumberPage + 1
+    const searchObject = parseSearchUrl(qs.parse(props.location.search))
+
+    const fetchGetSearchedKeywords = await props.dispatch(getSearchedKeywords(
+      searchObject.contentTypes,
+      props.searchResult.searchedKeywords,
+      NEXT_PAGE,
+      props.searchResult.numberResultsByPage,
+      searchObject.showArchived,
+      searchObject.showDeleted,
+      searchObject.showActive
+    ))
+
+    switch (fetchGetSearchedKeywords.status) {
+      case 200:
+        props.dispatch(setCurrentNumberPage(NEXT_PAGE))
+        props.dispatch(appendSearchResultsList(fetchGetSearchedKeywords.json.contents))
+        this.setState({ totalHits: fetchGetSearchedKeywords.json.total_hits })
+        props.history.push(PAGE.SEARCH_RESULT + '?' + qs.stringify({ ...qs.parse(props.location.search), p: NEXT_PAGE }, { encode: true }))
```

Comment:
I think this might be a bit of a misconception.

You are updating the local state then updating the route. It should be the other way around, the interface should react to the new route and trigger the new search when the route change.

Currently: click "see more" > trigger api GET search result > update result > update route for consistency
Expected: click "see more" > update route.
And the updated route triggers the rerender of the component that will GET the new search result

This is all theoretical, if it's an update of the old search process maybe it's too big of a change to do it now. But that is something important to keep in mind when building apps

Reply:
All this content part I just copy+paste from the simple search, I don't think it's too big to change it now (in both components), I'll fix it.